CrossPare/src/main/java/de/ugoe/cs/cpdp/loader/JitDataLoader.java

```diff
@@ -0,0 +1,255 @@
+	        }
+	    }
+	    temp[wordCount++] = line.substring(i); 
+	    String[] parts = new String[wordCount];
+	    System.arraycopy(temp, 0, parts, 0, wordCount);
+	    return parts;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see de.ugoe.cs.cpdp.loader.AbstractFolderLoader.SingleVersionLoader#load(
+	 * java.io.File)
+	 */
+	@Override
+	public Instances load(File file, boolean binaryClass) {
+		final String[] lines;
+		try {
+			List<String> stringList = Files.readAllLines(file.toPath());
+			lines = stringList.toArray(new String[] {});
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+
+		// configure Instances
+		final ArrayList<Attribute> atts = new ArrayList<>();
+
+		String[] lineSplit = lines[0].split(",");
+		List<Integer> usedAttributes = new ArrayList<>();
+		int committerDateIndex = 1;
+		int index = 3;
+		while (!"label_adhoc".equals(lineSplit[index])) {
+			// skip previous inducing as this was more of a test
+			if("previous_inducing".equals(lineSplit[index])) {
+				index++;
+				continue;
+			}
+			String curAttribute = lineSplit[index];
+			boolean hasMatch = false;
+        	for( String regex : attributeNames ) {
+        		if( curAttribute.matches(regex) ) {
+        			hasMatch = true;
+        		}
+        	}
+        	if( hasMatch || attributeNames.isEmpty() ) {
+        		atts.add(new Attribute(curAttribute));
+        		usedAttributes.add(index);
+        	}
+			index++;
+		}
+		
+		// set correct label index
+		int labelIndex = (adhoc) ? index : index + 1;
+		int issueMatrixIndex = index + 4;
+		Attribute classAtt;
+		if (binaryClass) {
+			// add nominal class attribute
+			final ArrayList<String> classAttVals = new ArrayList<>();
+			classAttVals.add("0");
+			classAttVals.add("1");
+			classAtt = new Attribute("bug", classAttVals);
+		} else {
+			// add numeric class attribute
+			classAtt = new Attribute("bugs");
+		}
+		atts.add(classAtt);
+		
+		final Instances data = new Instances(file.getName(), atts, 0);
+		data.setClass(classAtt);
+
+		// fetch data
+		for (int i = 1; i < lines.length; i++) {
+			lineSplit = split(lines[i], ',');;
+			double[] values = new double[usedAttributes.size()+1];
+			for (int j = 0; j < values.length - 1; j++) {
+				String value = lineSplit[usedAttributes.get(j)].trim();
+				// encode truth values to binary
+				if(!"True".equals(value) && !"False".equals(value)) {
+					values[j] = Double.parseDouble(lineSplit[usedAttributes.get(j)].trim());
+				} else {
+					values[j] = ("False".equals(value)) ? 0 : 1;
+				}
+			}
+			List<String> issueList = Arrays.asList(lineSplit[labelIndex].trim().split("\\s*,\\s*"));
+			if (binaryClass) {
+				// nominal class value
+				values[values.length - 1] = (issueList.size() == 0) ? 0 : 1;
+			} else {
+				// numeric class value
+				values[values.length - 1] = issueList.size();
+			}
+			data.add(new DenseInstance(1.0, values));
+		}
+		
+		// create issue matrix and fetch committer dates
```

Comment:
please separate issue matrix creation from time stamps. that makes the code easier to maintain and extend, the performance trade-off should be small